ls1intum/Helios
# feat: Add OpenAPI spec validation action and configure Spring profiles

### Motivation
Prevent manual changes to autogenerated client code and ensure REST endpoint updates are reflected in the OpenAPI YAML and client code automatically.

### Description
**Workflow Changes**:
- Added a GitHub Actions workflow named `Validate OpenAPI Spec and Generated Client Code`
  - Validates changes in openapi.yaml, server and client-generated code.
  - Automatically posts comments on pull requests when validation fails.
  - Removes sticky comments when validation passes.

**Gradle Changes**
- Added a new `bootRunDev` task to simplifies running the Spring Boot application in the `dev` profile
- Configured `openApi` Gradle plugin to set/use the `openapi` profile.
- Dynamically adds the H2 database dependency during the OpenAPI generation task (while generating the openapi yml file, the PostgreSQL connection is not necessary anymore).
- New task enforces an active Spring profile to be set before executing `bootRun`.

**Application server config changes**
- `application-dev.yml` added for local development
- `application-openapi.yml` added for openapi.yml generation task
- `application.yml` simplified for base usage

### Testing Instructions
- Run `./gradlew bootRunDev` to verify the `dev` profile works as intended.
- Execute `./gradlew generateOpenApiDocs` to ensure OpenAPI specs are generated correctly.
